server/src/test/java/org/apache/druid/server/coordinator/rules/ForeverLoadRuleTest.java

```diff
@@ -47,45 +51,90 @@ public void testSerdeNullTieredReplicants() throws Exception
   }
 
   @Test
-  public void testMappingNullTieredReplicants() throws Exception
+  public void testCreatingNegativeTieredReplicants()
   {
-    String inputJson = "{\n"
-                      + " \"type\": \"loadForever\"\n"
-                      + "}";
-    String expectedJson = "    {\n"
-                          + "      \"tieredReplicants\": {\n"
-                          + "        \"" + DruidServer.DEFAULT_TIER + "\": " + DruidServer.DEFAULT_NUM_REPLICANTS + "\n"
-                          + "      },\n"
-                          + "      \"type\": \"loadForever\"\n"
-                          + "    }";
-    ObjectMapper jsonMapper = new DefaultObjectMapper();
-    ForeverLoadRule inputForeverLoadRule = jsonMapper.readValue(inputJson, ForeverLoadRule.class);
-    ForeverLoadRule expectedForeverLoadRule = jsonMapper.readValue(expectedJson, ForeverLoadRule.class);
-    Assert.assertEquals(expectedForeverLoadRule.getTieredReplicants(), inputForeverLoadRule.getTieredReplicants());
+    MatcherAssert.assertThat(
+        Assert.assertThrows(DruidException.class, () ->
+            new ForeverLoadRule(
+                ImmutableMap.of(DruidServer.DEFAULT_TIER, -1),
+                null
+            )
+        ),
+        DruidExceptionMatcher.invalidInput().expectMessage(
+            Matchers.containsString(
+                StringUtils.format(
+                    "Invalid number of replicas for tier [%s]. Value [%d] must be positive.",
```

Comment:
Nit: In test assertions, you can use the final message with values interpolated, so that the reader knows the message that is being verified.
```
Invalid number of replicas for tier [_default_tier]. Value [-1] must be positive.
```